src/renderer/components/+user-management/+role-bindings/dialog.tsx

```diff
@@ -234,16 +245,32 @@ export class RoleBindingDialog extends React.Component<RoleBindingDialogProps> {
           isMulti
           themeName="light"
           placeholder="Select service accounts ..."
-          options={serviceAccountStore.items}
-          isOptionSelected={account => this.selectedAccounts.has(account)}
-          formatOptionLabel={account => (
+          options={this.serviceAccountOptions}
+          formatOptionLabel={option => (
             <>
               <Icon small material="account_box" />
-              {` ${account.getName()} (${account.getNs()})`}
+              {` ${option.label}`}
             </>
           )}
-          getOptionLabel={account => `${account.getName()} (${account.getNs()})`}
-          onChange={selected => this.selectedAccounts.replace(selected)}
+          onChange={(selected, meta) => {
+            switch (meta.action) {
+              case "clear":
+                this.selectedAccounts.clear();
+                break;
+              case "deselect-option":
+              case "remove-value":
+              case "pop-value":
+                if (meta.option) {
+                  this.selectedAccounts.delete(meta.option.value);
+                }
+                break;
+              case "select-option":
+                if (meta.option) {
+                  this.selectedAccounts.add(meta.option.value);
+                }
+                break;
+            }
+          }}
```

Comment:
This Open Closed Principle violation is also duplicated above.

Reply:
This thread was resolved without resolution.

Reply:
Resolved elsewhere now